docs/configure-artifact-repository.md

```diff
@@ -82,6 +82,16 @@ an access key, you will need to create a user with just the permissions you want
 to associate with the access key. Otherwise, you can just create an access key
 using your existing user account.
 
+Although users are not able to have a reduced scope, S3 access grants created
+via a user are. These credentials however, are temporary, and therefore need
+to be periodically generated/refreshed by an external mechanism.
```

Comment:
I originally put the note about reduced scope in here since the paragraph above addresses the scope of IAM users. I'm going to change this section in its entirety to just describe the bucket creation only and move the relevant credential setup into other subheadings.

Reply:
Yea I noticed that too, and yes I think moving it to a dedicated IAM user section while leaving the top as common bucket creation makes sense

Reply:
@agilgur5 Do you have any opinions on how much detail should be provided for each authentication option? The docs are currently rather thorough for IAM users, but basically very empty for IRSA. I've currently added a bit more detail for IRSA and IAM roles, but I'm unsure of how much to add for S3 access grants, since the setup is more involved. I'd prefer to defer to the [AWS documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-grants-get-started.html), and just say once you have obtained credentials in step 4, you can use them in Argo Workflows with the snippet below

Reply:
Yes generally we should defer & link to upstream docs where possible, use examples, and avoid too much commentary. "Simple and direct language", per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language), makes for very efficient and straightforward docs.

Best practices (like security recommendations on this page) are good to note, but we don't need to go into detail for those, as they are not Argo specific and so are better explained upstream anyway.

Reply:
Ok great. I'll push up what I'm proposing now and would appreciate if you could let me know if I should continue in this direction